new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Domain.java

```diff
@@ -292,4 +324,231 @@ public static boolean scaleClusterWithRestApi(String domainUid,
+   * @throws IOException if an I/O error occurs
+   * @throws InterruptedException if any thread has interrupted the current thread
+   */
+  public static boolean scaleClusterWithWLDF(String clusterName,
+                                             String domainUid,
+                                             String domainNamespace,
+                                             String domainHomeLocation,
+                                             String scalingAction,
+                                             int scalingSize,
+                                             String opNamespace,
+                                             String opServiceAccount,
+                                             String myWebAppName,
+                                             String curlCommand)
+      throws ApiException, IOException, InterruptedException {
+
+    // create RBAC API objects for WLDF script
+    logger.info("Creating RBAC API objects for WLDF script");
+    if (!createRbacApiObjectsForWLDFScript(domainNamespace, opNamespace)) {
+      return false;
+    }
+
+    // copy scalingAction.sh to Admin Server pod
+    // NOTE: you must copy scalingAction.sh to $DOMAIN_HOME/bin/scripts on admin server pod
+    String adminServerPodName = domainUid + "-" + ADMIN_SERVER_NAME_BASE;
+    V1Pod adminPod = Kubernetes.getPod(domainNamespace, null, adminServerPodName);
+    if (adminPod == null) {
+      logger.info("The admin pod {0} does not exist in namespace {1}!", adminServerPodName, domainNamespace);
+      return false;
+    }
+
+    // create $DOMAIN_HOME/bin/scripts directory on admin server pod
+    logger.info("Creating directory {0}/bin/scripts on admin server pod", domainHomeLocation);
+    ExecResult result = exec(adminPod, null, true,
+        "/bin/sh", "-c", "mkdir -p " + domainHomeLocation + "/bin/scripts");
+    if (result.exitValue() != 0) {
+      return false;
+    }
+
+    logger.info("Copying scalingAction.sh to admin server pod");
+    try {
+      copyFileToPod(domainNamespace, adminServerPodName, null,
+          Paths.get(PROJECT_ROOT + "/../src/scripts/scaling/scalingAction.sh"),
+          Paths.get(domainHomeLocation + "/bin/scripts/scalingAction.sh"));
+    } catch (ApiException apex) {
+      logger.severe("Got ApiException while copying file to admin pod {0}", apex.getResponseBody());
+      return false;
+    } catch (IOException ioex) {
+      logger.severe("Got IOException while copying file to admin pod {0}", ioex.getStackTrace());
+      return false;
+    }
+
+    logger.info("Adding execute mode for scalingAction.sh");
+    result = exec(adminPod, null, true,
+        "/bin/sh", "-c", "chmod +x " + domainHomeLocation + "/bin/scripts/scalingAction.sh");
+    if (result.exitValue() != 0) {
+      return false;
+    }
+
+    // copy wldf.py and callpyscript.sh to Admin Server pod
+    logger.info("Copying wldf.py and callpyscript.sh to admin server pod");
```

Comment:
I don't understand the flow here. Why are we exec'ing a scaling action and then also calling a wldf script? I don't understand the use case being tested.

Reply:
We need to create WLDF policy rule and action through wlst. wldf.py is to create the WLDF Policy rule and set the script action to call scalingAction.sh.